### ISSUE TITLE
Compiler incorrectly compiling pointer constraints

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn#40182]
Consider this code

```cs
unsafe class Generic<T> where T : byte*
{
}
```

This should issue the following error:

> Program.cs(10,39): error CS0706: Invalid constraint type. A type used as a constraint must be an interface, a non-sealed class or        a type parameter.

However the code is instead compiling. 

Checked with the native csc compiler and it does issue this error.
